lib/widgets/path_editor/editors/graph_editor.dart

```diff
@@ -0,0 +1,427 @@
+        top: BorderSide(color: Colors.white24, width: 4),
+      ),
+    );
+
+    LineChartData data = LineChartData(
+      lineTouchData: LineTouchData(enabled: false),
+      gridData: FlGridData(show: true),
+      titlesData: titlesData,
+      borderData: borderData,
+      lineBarsData: lineBarsData,
+      extraLinesData: extraLines,
+      minX: 0,
+      maxX: time,
+      minY: min.floorToDouble(),
+      maxY: max.ceilToDouble(),
+    );
+
+    return LineChart(
+      data,
+      swapAnimationDuration: const Duration(milliseconds: 0),
+    );
+  }
+
+  String markerLabel(VerticalLine line) {
+    for (int idx = 0; idx < path.markers.length; idx++) {
+      if (path.markers[idx].timeSeconds == line.x) {
+        return path.markers[idx].name;
+      }
+    }
+    return '';
+  }
+
+  Widget leftTitleWidgets(double value, TitleMeta meta) {
+    if ((value - value.toInt()).abs() > 0.01) {
+      return const Text('');
+    }
+
+    const style = TextStyle(
+      color: Colors.white,
+      fontWeight: FontWeight.bold,
+      fontSize: 14,
+    );
+
+    return SideTitleWidget(
+        axisSide: meta.axisSide,
+        space: 10,
+        child: Text(value.toInt().toString(),
+            style: style, textAlign: TextAlign.center));
+  }
+
+  Widget bottomTitleWidgets(double value, TitleMeta meta) {
+    if (((value * 10.0) - (value * 10.0).round().toInt()).abs() > 0.01) {
+      return const Text('');
+    }
+
+    const style = TextStyle(
+      color: Colors.white,
+      fontWeight: FontWeight.bold,
+      fontSize: 16,
+    );
+
+    return SideTitleWidget(
+      axisSide: meta.axisSide,
+      space: 10,
+      child: Text(value.toStringAsFixed(1), style: style),
+    );
+  }
+}
+
+class GraphEditor extends StatefulWidget {
+  final RobotPath path;
+  final bool holonomicMode;
+  final void Function(RobotPath path) savePath;
+  final SharedPreferences prefs;
+  static String prefShowVelocity = 'showVelocity';
+  static String prefShowAccel = 'showAccel';
+  static String prefShowHeading = 'showHeading';
+  static String prefShowRotation = 'showRotation';
+  static String prefShowAngularVelocity = 'showAngularVelocity';
+  static String prefShowCurvature = 'showCurvature';
+  static Color colorVelocity = Colors.red;
+  static Color colorAccel = Colors.orange;
+  static Color colorHeading = Colors.yellow;
+  static Color colorRotation = Colors.yellow;
+  static Color colorAngularVelocity = Colors.green;
+  static Color colorCurvature = Colors.blue;
+
+  const GraphEditor(
+      {required this.path,
+      required this.holonomicMode,
+      required this.savePath,
+      required this.prefs,
+      super.key});
+
+  @override
+  State<GraphEditor> createState() => _GraphEditorState();
+}
+
+class _GraphEditorState extends State<GraphEditor> {
+  final GlobalKey _key = GlobalKey();
+  UniqueKey _graphRuntimeKey = UniqueKey();
+  bool isSampled = false;
+  bool cardMinimized = false;
```

Comment:
Prefix with `_` to keep these private unless they're accessed from another class. Not really a huge deal its just a bit more "correct"

```suggestion
  bool _isSampled = false;
  bool _cardMinimized = false;
```